### DISCUSSION TITLE
Normalized perpendicular

### DISCUSSION THREAD

**Discussion by bstro** [DerYeger/vecti#18] [Q&A] [answered]
Hi there, Vecti looks fantastic. I'm trying to adapt this [stackoverflow suggestion](https://stackoverflow.com/a/57066596) to your API:

![image](https://user-images.githubusercontent.com/627290/187829503-60aa2098-5aeb-4706-9027-981e65e74663.png)

What I'm trying to achieve: Given line AB, how do I calculate points C and D?

So far, I have:
```
const A = new Vector(x1, y1);
const B = new Vector(x2, y2);
const AB = B.subtract(A);
```

My question is, how can I calculate the normalized perpendicular of line `AB`? Its been a long since precalculus :D

**Comment by DerYeger** [comment, marked as answer]
Hi, this shouldn't be too difficult!

You could use [this](https://gamedev.stackexchange.com/a/70076) StackOverflow answer, or use [Vecti's methods](https://vecti.yeger.eu/guide/) as seen below:

```ts
const b = new Vector(x1, y1);
const c = new Vector(x2, y2);
const aToB = B.subtract(A);

const orth = aToB.rotateByDegrees(90).normalize()

// Multiple by (-1) * 6 / 2 to get C, 6 / 2 to get D
const c = b.add(orth.multiply(-3))
const d = b.add(orth.multiply(3))
```

Let me know if this works, I wrote it off the top of my head.

**Reply by bstro** [reply]
That did it! Thanks a bunch! 🥇

**Reply by DerYeger** [reply]
> That did it! Thanks a bunch! 🥇 

Glad I could help!
